### PR TITLE
Fix --stream-errors w/o --stream, and doc+test (close #2748)

### PR DESCRIPTION
**Pull request by nicowilliams**

### CODE REVIEW

**Comment by nicowilliams** [comment]
~Still need to add a test.~

**Comment by nicowilliams** [comment]
Now with a [minimal] test.

File: tests/shtest [hunk 1]
```diff
@@ -180,6 +180,11 @@ $VALGRIND $Q $JQ -c '. as $d|path(..) as $p|$d|getpath($p)|select((type|. != "ar
 $VALGRIND $Q $JQ --stream -c '.|select(length==2)' < "$JQTESTDIR/torture/input0.json" > $d/out1
 diff $d/out0 $d/out1
 
+printf '["Unfinished JSON term at EOF at line 1, column 1",[0]]\n' > $d/expected
+printf '[' | $VALGRIND $Q $JQ --stream-errors -c . > $d/out 2>&1
+diff $d/out0 $d/out1
```
Comment (itchyny) [inline tests/shtest]: Compare `$d/expected` and `$d/out`.

Comment (nicowilliams) [reply tests/shtest]: Oops.  Fixed.

File: docs/content/manual/manual.yml [hunk 2]
```diff
@@ -130,6 +130,16 @@ sections:
...
 
+      * `--stream-errors`:
+
+        Like `--stream`, but invalid JSON inputs yield array calues
+        where the first element is the error and the second is a path.
+        For example, `["a",n]` produces ["Invalid literal at line 1,
+        column 9",[1]]`.
+
+        Implies `--stream`.  Invalid JSON inputs produce no error values
+        when `--stream` w/o `--stream-errors`.
```
Comment (itchyny) [inline docs/content/manual/manual.yml]: I prefer to write "without" to make it read explicit.

**Review by itchyny — APPROVED** [review]
LGTM!

**Pull request merged**